### PR TITLE
bump: kubernetes client version

### LINKED ISSUE #36678: apache-airflow-providers-cncf-kubernetes requires latest kubernetes version
### Description

Update apache-airflow-providers-cncf-kubernetes to use kubernetes latest versions 
kubernetes | >=21.7.0,<24
-- | --

### Use case/motivation

Kubernetes has released latest versions recently v1.29 but airflow kubernetes is stuck with v1.23 and not compatible with the latest versions.
https://github.com/kubernetes-client/python

### Related issues

Since the kubernetes v1.23 has come to end of life and eks is not supporting the version. It would be nice to see upgraded
 apache-airflow-providers-cncf-kubernetes

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by raphaelauv**
close: https://github.com/apache/airflow/issues/36678

### CODE REVIEW

File: airflow/providers/cncf/kubernetes/provider.yaml [hunk 1]
```diff
@@ -88,13 +88,13 @@ dependencies:
...
   # core is not hard-limited via install-requirements, only by extra).
-  - kubernetes>=21.7.0,<24
+  - kubernetes>=23.6.0,<29
   # The Kubernetes_asyncio package is used for providing Asynchronous (AsyncIO) client library for
   # standard Kubernetes API. The version is limited by minimum 18.20.1 because of introducing the ability to
   # load kubernetes config file from dictionary in that release and is limited to the next MAJOR version
   # (started from current 24.2.2 version) to prevent introducing some problems that could be due to some
   #  major changes in the package.
-  - kubernetes_asyncio>=18.20.1,<25
+  - kubernetes_asyncio>=23.6.0,<29
```
Comment (eladkal) [inline airflow/providers/cncf/kubernetes/provider.yaml:97]: The above comment blocks refers to 18.20.1
It needs to be modified

**Comment by potiuk** [comment]
Hmm @raphaelauv - are you planning to update it ? Seems that we do need to bump min-version of the k8s client - see 
 #37008

File: generated/provider_dependencies.json [hunk 2]
```diff
@@ -280,8 +280,8 @@
       "asgiref>=3.5.2",
       "cryptography>=2.0.0",
       "google-re2>=1.0",
-      "kubernetes>=21.7.0,<24",
-      "kubernetes_asyncio>=18.20.1,<25"
+      "kubernetes>=23.6.0,<29",
```
Comment (dirrao) [inline generated/provider_dependencies.json:283]: 23.6.0 is breaking with airflow 2.8.1. We need to bump the minimum Kubernetes version to be  28.1.0.
#37008. [Bug fix](https://github.com/kubernetes-client/python/pull/2095)

**Comment by potiuk** [comment]
> @potiuk I'm not available to finish this PR in the next days , should I close it ?

cc: @dirrao -> maybe you would like to take that over and open a new one?

**Comment by dirrao** [comment]
> > @potiuk I'm not available to finish this PR in the next days , should I close it ?
> 
> cc: @dirrao -> maybe you would like to take that over and open a new one?

sure

**Pull request closed without merging**